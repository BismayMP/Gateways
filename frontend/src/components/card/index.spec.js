import { cleanup, render, screen } from '@testing-library/react'
import Card from './index'
import renderer from 'react-test-renderer'

afterEach(cleanup)

describe('Card component', () => {
  const gateway = { name: 'test gateway', ipv4: '1.1.1.1' }
  const onDelete = jest.fn((item) => null)
  const showDetails = jest.fn((item) => null)
  test('should match snapshot', () => {
    const component = renderer.create(
      <Card gateway={gateway} onDelete={onDelete} showDetails={showDetails} />,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('All elements are in the dom', () => {
    render(
      <Card gateway={gateway} onDelete={onDelete} showDetails={showDetails} />,
    )
    const buttons = screen.getAllByRole('button')
    buttons.forEach((button) => {
      expect(button).toBeInTheDocument()
      button.click()
    })
    expect(onDelete).toHaveBeenCalledTimes(1)
    expect(showDetails).toHaveBeenCalledTimes(1)
    const name = screen.getByText(gateway.name)
    expect(name).toBeInTheDocument()
    const ipv4 = screen.getByText(gateway.ipv4)
    expect(ipv4).toBeInTheDocument()
    const devicesAmount = screen.getByText('Peripheral devices: 0')
    expect(devicesAmount).toBeInTheDocument()
  })
})
