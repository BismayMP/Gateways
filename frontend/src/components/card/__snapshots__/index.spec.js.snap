// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should match snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root card-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root card-title MuiTypography-h5 MuiTypography-colorTextPrimary"
    >
      test gateway
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      1.1.1.1
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      Peripheral devices:
       
      0
    </p>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Go to details
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Add device
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Remove Gateway
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
